src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/TempDirectoryWorkloadManifestProvider.cs

```diff
@@ -4,27 +4,46 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Linq;
 
 namespace Microsoft.NET.Sdk.WorkloadManifestReader
 {
     public class TempDirectoryWorkloadManifestProvider : IWorkloadManifestProvider
     {
-        private readonly string _sdkVersionBand;
         private readonly string _manifestsPath;
+        private readonly string _sdkVersionBand;
 
+        /// <param name="manifestsPath">
+        ///     Result of directly extract manifest NuGet Packages. Should contain folders like
+        ///     microsoft.net.workload.emscripten.manifest-6.0.100.6.0.0-preview.7.21377.2
+        /// </param>
         public TempDirectoryWorkloadManifestProvider(string manifestsPath, string sdkVersion)
         {
             _manifestsPath = manifestsPath;
             _sdkVersionBand = sdkVersion;
         }
 
-        public IEnumerable<(string manifestId, string? informationalPath, Func<Stream> openManifestStream)> GetManifests()
+        public IEnumerable<(string manifestId, string? informationalPath, Func<Stream> openManifestStream)>
+            GetManifests()
         {
             foreach (var workloadManifestDirectory in GetManifestDirectories())
             {
-                var workloadManifest = Path.Combine(workloadManifestDirectory, "WorkloadManifest.json");
-                var id = Path.GetFileName(workloadManifestDirectory);
-                yield return (id, workloadManifest, () => File.OpenRead(workloadManifest));
+                string? workloadManifest = Path.Combine(workloadManifestDirectory, "WorkloadManifest.json");
+                FileInfo? nupkg = new DirectoryInfo(workloadManifestDirectory).GetFiles("*.nuspec",
+                    SearchOption.TopDirectoryOnly).SingleOrDefault();
+                if (nupkg == default)
+                {
+                    throw new Exception($"Package {nupkg} contains more than one nuspec");
+                }
+
+                string? packageId = Path.GetFileNameWithoutExtension(nupkg.FullName);
+                if (packageId.Contains("-"))
+                {
+                    int index = packageId.LastIndexOf("-", StringComparison.OrdinalIgnoreCase);
+                    packageId = packageId.Substring(0, index);
```

Comment:
Does this work as intended for complex versions? For example, `microsoft.net.workload.emscripten.manifest-6.0.100.6.0.0-preview.7.21377.2` would become `microsoft.net.workload.emscripten.manifest-6.0.100.6.0.0`, but we want to remove the `.6.0.0` as well, right?

Reply:
turned out the right workload ID is everything after .manifest: microsoft.net.workload.emscripten**.manifest**-6.0.100.6.0.0 -> microsoft.net.workload.emscripten

So we won't have this issue. I changed the logic in the new push